store/mongo/datastore_mongo.go

```diff
@@ -1912,3 +1920,59 @@ func (db *DataStoreMongo) ExistByArtifactId(ctx context.Context,
 
 	return true, nil
 }
+
+// Per-tenant storage settings
+func (db *DataStoreMongo) GetStorageSettings(ctx context.Context, tenantID string) (*model.StorageSettings, error) {
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collection := database.Collection(CollectionStorageSettings)
+
+	settings := new(model.StorageSettings)
+	// supposed that it's only one document in the collection
+	query := bson.M{
+		"_id": bson.M{
+			"$exists": true,
+		},
+	}
+	if err := collection.FindOne(ctx, query).Decode(settings); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return settings, nil
+}
+
+func (db *DataStoreMongo) SetStorageSettings(ctx context.Context, storageSettings *model.StorageSettings, tenantID string) error {
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collection := database.Collection(CollectionStorageSettings)
+
+	// need to make sure that it's only one document exists in the collection
+	// query checks if a document exists in a collection and
+	// with enabled upsert it will be created or overwritten
+	filter := bson.M{
+		"_id": bson.M{
+			"$exists": true,
+		},
```

Comment:
Another option is using a fix value for ID (e.g. `settings`), without relying on the autogeneration of an objectID by mongo. In this case, you could use `$setOnInsert` in the update document to set the `_id` to the constant value above.

Reply:
@tranchitella you requested changes. Is the request is related to model and this is optional? or it would be good to change the query in datastore as well?

Reply:
@tranchitella not really able to get it working and based on the following error message it seems like it should not work...
```
internal error: Failed to save settings: multiple write errors: [{write errors: [{Performing an update on the path '_id' would modify the immutable field '_id'}]}, {<nil>}]
```

Reply:
finally got it working :)